### DISCUSSION TITLE
Is there a way to copy only part of text in list view item?

### DISCUSSION THREAD

**Discussion by En3Tho** [tui-cs/Terminal.Gui#1700] [General]
I've written a small GUI for dotnet-dump analyze
It basically works with text provided from dotnet-dump process and adds it to new tab with own list view and filter field

![image](https://user-images.githubusercontent.com/37334640/167258579-c48ae359-ea9a-4c7f-9dd6-9ab46ecc21fc.png)

Is there a way to select/copy only a part of text in selected list view item? E.g. I need only the address. That would be very convinient.
List view is used for result filtration and it looks quite neat, I would rather not change it

**Comment by BDisp** [comment]
I think not, but if can use a custom class or record with the necessary properties and override the `ToString` method to display what you want. Then the `ListView` source will be a `List<yourClassOrRecord>`. With this situation you can create a method to copy the address that will get the `Address` property of your custom class or record.

**Reply by En3Tho** [reply]
I see. The problem is that I want to keep it simple and just use text provided by dotnet-dump. Or else I will have to write parsers for every possible SOS command. It will be a full fledged application then. Too much work and I'm not sure it's worth it.

**Reply by En3Tho** [reply]
I guess a different kind of ListView is needed here - with custom views as items. Or an existing one needs something like a RenderRow (obj -> View) factory method. I will try looking into that but it seems to be quite a lot to implement too.
Upd: It seems like IListViewDataSource already has this method. I will try to look into it then.

**Reply by En3Tho** [reply]
In the end I've taken the parser path instead of custom views. Just making dialog with available subcommands like this: 
![image](https://user-images.githubusercontent.com/37334640/167821873-21e28a78-78a9-42b6-b716-4847aed08872.png)
Again, thank you for this framework. It's cool :)

**Reply by BDisp** [reply]
Excellent work. You can also use the `ContextMenu` to run the commands.

**Reply by En3Tho** [reply]
I will try, thanks. By the way, I'd you're interested, any contribution is welcome :p
